stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -1583,8 +1583,17 @@ extension ${Self} {
 % for self_ty in all_integer_types(word_bits):
 %   That = self_ty.stdlib_name
 %   ThatBuiltinName = self_ty.builtin_name
-%   srcBits = self_ty.bits
 %   sign = 's' if self_ty.is_signed else 'u'
+%   srcBits = self_ty.bits
+%   dstBits = SignificandBitCount + 1 # implicit leading one
+%   if self_ty.is_signed:
+%     # The magnitude of the minimum value of a signed integer type is always
+%     # representable in floating point (assuming a large enough exponent)
+%     # because only one bit is set. All other possible magnitudes can be
+%     # represented using one bit less than the original source type.
+%     dstBits += 1
```

Comment:
Can we simplify this to clarify the logic here? It's not `dstBits` that gains one if the source type is signed; rather, the maximum representable positive source value uses one fewer bit if the source is signed. Also, the implicit leading one can remain implicit if we continue to use `SignificandBitCount` explicitly:

```swift
%  srcBits = self_ty.bits # No change.
%  srcBitsExcludingSign = srcBits - 1 if self_ty.is_signed else srcBits
%  # No need to define `dstBits`.

/* ... */
```

We can also tighten up the logic so as to check, rather than assume, a sufficient number of exponent bits:

```swift
%    if srcBitsExcludingSign < SignificandBitCount and srcBits < 2 ** (ExponentBitCount - 1):
  @available(*, message: "Converting ${That} to ${Self} will always succeed.")
/* ... */
```

Reply:
I'd like to keep the implicit bit in the count. It makes the mental juggling easier in my opinion. You're right that it shouldn't really be called dstBits though. I'll try renaming it to something like maxBitLength that makes the intent a bit clearer.

I'll add the exponent size check to the conditions.